packages/react-router/docs/guides/testing.md

```diff
@@ -158,43 +107,43 @@ const App = () => (
     />
   </div>
 );
+```
 
-// the actual test!
-it("navigates around", done => {
-  renderTestSequence({
-    // tell it the subject you're testing
-    subject: App,
-
-    // and the steps to execute each time the location changes
-    steps: [
-      // initial render
-      ({ history, div }) => {
-        // assert the screen says what we think it should
-        console.assert(div.innerHTML.match(/Welcome/));
-
-        // now we can imperatively navigate as the test
-        history.push("/dashboard");
-      },
-
-      // second render from new location
-      ({ div }) => {
-        console.assert(div.innerHTML.match(/Dashboard/));
-
-        // or we can simulate clicks on Links instead of
-        // using history.push
-        Simulate.click(div.querySelector("#click-me"), {
-          button: 0
-        });
-      },
-
-      // final render
-      ({ location }) => {
-        console.assert(location.pathname === "/");
-        // you'll want something like `done()` so your test
-        // fails if you never make it here.
-        done();
-      }
-    ]
+```jsx
+// app.test.js
+it("navigates home when you click the logo", async => {
+  // in a real test a renderer like "@testing-library/react"
+  // would take care of setting up the DOM elements
+  const root = document.createElement('div')
+  document.body.appendChild(root);
+  
+  // Set initial location
+  window.history.pushState({}, '', '/my/initial/route');
+  
+  // Render app
+  render(
+    <BrowserRouter>
+      <App />
+    <BrowserRouter>,
+    root
+  )
+  
+  // Interact with page
+  act(() => {
+    // Find the link (perhaps using the text content)
+    const goHomeLink = document.querySelector('#nav-logo-home');
+    // Click it
+    goHomeLink.dispatchEvent(new MouseEvent("click", { bubbles: true }));
   });
+  
+  // Assert about new location
+  expect(window.location.pathname).toBe('/homepage');
+  
+  // Check correct page content showed up
+  expect(document.body.textContent).toBe('Home')
 });
 ```
+
```

Comment:
@pshrmn how about adding a note like this?

```suggestion

### Resetting location between tests

To make tests less dependent on the order of the tests, navigate to a known location between tests when using BrowserRouter.

```js
beforeEach(() => window.history.pushState({}, '', startUrl))
```

### BrowserRouter vs Alternatives

Instead of `BrowserRouter`, you can also use one of the other router components:

- `MemoryRouter` if you don't need to navigate or access the history object, and don't need `window.location` to be updated by the router.
- The base `Router` with a custom `history` object

The advantage of these these is that you can easily reset the router between tests, and don't need to rely on the `window.history` object.

```